fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragmentStore.kt

```diff
@@ -139,6 +139,16 @@ sealed class HistoryFragmentAction : Action {
      */
     data class HistoryItemLongClicked(val item: History) : HistoryFragmentAction()
 
+    /**
+     * The user has indicated that a time range of history items should be deleted.
+     */
+    data class DeleteTimeRange(val timeFrame: RemoveTimeFrame?) : HistoryFragmentAction()
```

Comment:
I actually think it would make the most sense to add a `RemoveTimeFrame.AllTime` subtype, but I'm not sure that's within the scope of this ticket. Happy to file a follow-up or sneak it in, reviewer's choice

Reply:
Basically I think that the logic [here](https://searchfox.org/mozilla-mobile/rev/d9485672d81b7accf4a1561441024309936b0a0c/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt#441) is a little odd for the same reason - `null` doesn't communicate semantic meaning to me very well in this case.

Reply:
I'm cool with a follow-up. Either approach sounds good

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1848057